sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/metadata/datasource/dialect/OracleDataSourceMetaData.java

```diff
@@ -45,11 +45,19 @@
     
     public OracleDataSourceMetaData(final String url) {
         Matcher matcher = pattern.matcher(url);
-        if (!matcher.find()) {
+        if (matcher.find()) {
+            hostName = matcher.group(3);
+            port = Strings.isNullOrEmpty(matcher.group(4)) ? DEFAULT_PORT : Integer.valueOf(matcher.group(4));
+            schemaName = matcher.group(5);
+        } else if (url.contains("HOST") && url.contains("PORT") && url.contains("SERVICE_NAME")) {
```

Comment:
Could you use reg expression to process connection URL? just keep same way with previous codes.

Reply:
Agree